kernel/src/fs/procfs/pid/stat.rs

```diff
@@ -1,20 +1,28 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use core::fmt::Write;
+use core::{fmt::Write, sync::atomic::Ordering};
 
 use crate::{
     fs::{
         procfs::template::{FileOps, ProcFileBuilder},
         utils::Inode,
     },
     prelude::*,
+    process::posix_thread::AsPosixThread,
+    thread::Thread,
+    vm::vmar::RssType,
     Process,
 };
 
-/// Represents the inode at `/proc/[pid]/stat`.
-/// The fields are the same as the ones in `/proc/[pid]/status`. But the format is different.
-/// See https://github.com/torvalds/linux/blob/ce1c54fdff7c4556b08f5b875a331d8952e8b6b7/fs/proc/array.c#L467
-/// FIXME: Some fields are not implemented yet.
+/// Represents the inode at either `/proc/[pid]/stat` or `/proc/[pid]/task/[tid]/stat`.
+///
+/// If `is_pid_stat` is true, it corresponds to the process-level `/proc/[pid]/stat`.
```

Comment:
Do not refer to any private fields in Rust docs. The users of a struct should not be aware of implementation details.